services/construction.go

```diff
@@ -17,6 +18,7 @@ import (
 	"github.com/oasisprotocol/oasis-core/go/common/quantity"
 	"github.com/oasisprotocol/oasis-core/go/consensus/api/transaction"
 	staking "github.com/oasisprotocol/oasis-core/go/staking/api"
+	"github.com/tendermint/tendermint/mempool"
```

Comment:
This sounds like a very bad idea, this should be done in Oasis Core and exposed as a `ErrDuplicateTx`, not exposing internal details. I would rather you just hardcode the error string here and add an `XXX` comment below, as the used version of Tendermint (added to `go.mod` above) is inconsistent with what Oasis Core is actually using anyway.

Reply:
yeah, it is uncool to add such a big module as a direct dependency for a constant value

let's just wait for another release of oasis-core. whether this call succeeds or fails isn't affecting anything by much. it's still the rosetta client's responsibility to rebroadcast once in a while even if it's indicated that it succeeds

Reply:
we have now waited for that